runtime/server/jwt/interceptors.go

```diff
@@ -0,0 +1,101 @@
+package jwt
+
+import (
+	"context"
+	"strings"
+
+	"github.com/golang-jwt/jwt"
+	grpc_middleware "github.com/grpc-ecosystem/go-grpc-middleware"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+)
+
+type jwtContextKey struct{}
+
+// JWT secret for signing and verifying tokens
+var jwtSecret = []byte("secret")
+
+// Claims represents the payload of a JWT
+type Claims struct {
+	InstanceID string `json:"instance_id"`
+	jwt.StandardClaims
+}
+
+// UnaryServerInterceptor is a middleware for parsing and extracting JWT from metadata
+func UnaryServerInterceptor(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+	newCtx, err := parseJWTAndPopulateContextWithClaims(ctx)
+	if err != nil {
+		return nil, err
+	}
+	return handler(newCtx, req)
+}
+
+func StreamServerInterceptor(srv interface{}, ss grpc.ServerStream, info *grpc.StreamServerInfo, handler grpc.StreamHandler) error {
+	newCtx, err := parseJWTAndPopulateContextWithClaims(ss.Context())
+	if err != nil {
+		return err
+	}
+	wrapped := grpc_middleware.WrapServerStream(ss)
+	wrapped.WrappedContext = newCtx
+
+	return handler(srv, wrapped)
+}
+
+func parseJWTAndPopulateContextWithClaims(ctx context.Context) (context.Context, error) {
```

Comment:
This should only error if there's invalid/malformed/expired auth info in the headers (things like an `authorization` header without a token), but not when no auth info is provided. When no auth info is provided, it should instead return successfully (populating the ctx with claims corresponding to an anon request).

Reply:
this also should depend on the env in case of no auth info ? if prod then it will be empty claims if local then all claims ?